storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -747,4 +807,24 @@ private void putBlock(final SignedBeaconBlock block) {
       writeLock.unlock();
     }
   }
+
+  @VisibleForTesting
+  Optional<Set<StateAndBlockSummary>> getEpochStates() {
+    return maybeEpochStates;
+  }
+
+  void removeStateAndBlock(final Bytes32 root) {
+    blocks.remove(root);
+    states.remove(root);
+  }
```

Comment:
This probably should remove from epoch states too which would work well if it's keyed by block root (not so well if it winds up being the checkpoint state cache I mention above...). In long periods of non-finalisation you may have some relatively long chains invalidated when finalisation invalidates a fork.

Reply:
that was my initial intent, it would have been so simple...
pruned hot blocks listed the finalized block so it deleted the associated state, so i handled the epochs separately initially so that it remains...